### DISCUSSION TITLE
can't Show selected Image In UIImageView

### DISCUSSION THREAD

**Discussion by WorkWithKaranMehra** [mikaoj/BSImagePicker#327] [Show and tell]
{
 let vc = ImagePickerController()
            vc.settings.selection.max = 10

        presentImagePicker(vc, animated: true,
                select: { (asset: PHAsset) -> Void in
                    print("Selected: \(asset)")
                }, deselect: { (asset: PHAsset) -> Void in
                    print("Deselected: \(asset)")
                }, cancel: { (assets: [PHAsset]) -> Void in
                    print("Cancel: \(assets)")
                }, finish: { (assets: [PHAsset]) -> Void in
                    print("Finish: \(assets)")
                }, completion: nil)
    }
    
    func imagePickerController(picker: ImagePickerController, didFinishPickingImage image: UIImage!, editingInfo: [NSObject : AnyObject]!) {
        print("image selected from gallery 22")
        self.imageView.image = image
        print("image selected from gallery")
        self.dismiss(animated: true, completion: nil)
    
}

It selected The Image But Not Show Please Help.....
#Swift5

**Comment by mikaoj** [comment]
> It selected The Image But Not Show Please Help.....

I didn't get that, could you try to explain it a bit more?

**Reply by WorkWithKaranMehra** [reply]
okay

Today I make a Project which Is Pick a photo into a gallery and show In UIImageView
1. Image selected to the gallery successfully
2. when I click Done. Image Not Show In UIImageView

**Reply by WorkWithKaranMehra** [reply]
![Screen Shot 2021-03-03 at 12 02 49 PM](https://user-images.githubusercontent.com/73272327/109763777-97a27d80-7c18-11eb-9a87-671848e6d0f8.png)

**Reply by mikaoj** [reply]
Ah alright. The image picker has 2 ways of doing callbacks. The easiest to use it the closure way, but for more advanced use-cases you can implement the delegate style.

You seem to have implemented almost a UIImagePickerDelegate method? That doesn't work.
remove the ```func imagePickerController(picker: ImagePickerController, didFinishPickingImage image: UIImage!, editingInfo: [NSObject : AnyObject]!) {``` method.
And add code in the finish closure.

```swift
    presentImagePicker(vc, animated: true,
            select: { (asset: PHAsset) -> Void in
                print("Selected: \(asset)")
            }, deselect: { (asset: PHAsset) -> Void in
                print("Deselected: \(asset)")
            }, cancel: { (assets: [PHAsset]) -> Void in
                print("Cancel: \(assets)")
            }, finish: { (assets: [PHAsset]) -> Void in
                // This closure will be called when a user presses done.
                // Convert the selected assets to UIImages of appropriate size
                //
            }, completion: nil)
```

**Reply by mikaoj** [reply]
There is an example in the README of how you fetch an image for an asset.

**Reply by WorkWithKaranMehra** [reply]
soory but. Its already used its not work

**Reply by WorkWithKaranMehra** [reply]
https://user-images.githubusercontent.com/73272327/109764326-6b3b3100-7c19-11eb-89c1-3352a3622a1b.mov

**Reply by mikaoj** [reply]
I know that it doesn't work. Because the code you showed have no chance of actually working.

1. Remove the UIImagePickerDelegate method where you try to set the selected image in the image view. This will never be called as this is BSImagePicker and not UIImagePicker.
2. Add code in the finish closure when you are presenting the image picker. That code should take the given assets, convert them to images of desired size and then set on your image view.

**Reply by WorkWithKaranMehra** [reply]
![Screen Shot 2021-03-03 at 12 14 43 PM](https://user-images.githubusercontent.com/73272327/109764860-12b86380-7c1a-11eb-98d3-73062ff163cf.png)

**Reply by mikaoj** [reply]
Awesome, that is step 1.
Now you need to add code in the finish closure that.
1. Converts the assets to images (see exampe code on how to do that in README)
2. Take those image and set on your image views (s)
